### PR TITLE
Add a paragraph about the reasons why limiting styles.

### LINKED ISSUE #373: Wide Review Comment 2017: future styling options mechanism
Copy/paste from https://lists.w3.org/Archives/Public/public-tt/2017Sep/0080.html - raising as an issue for tracking/disposition purposes.

The list of ::cue CSS properties in §7.2.1 is a useful minimal set, especially for a non-CSS-based implementation however the requirement to ignore any properties not in the list seems overly restrictive for a CSS based implementation. What is the mechanism for adding further styling options in the future, given that WebVTT files do not indicate any specification version?

### PR DESCRIPTION
**Pull request by silviapfeiffer**
closes #373

### CODE REVIEW

**Comment by silviapfeiffer** [comment]
@dwsinger need a review

File: index.bs [hunk 1]
```diff
@@ -302,6 +302,11 @@ one line except when a line break is definitely necessary.</p>
 target WebVTT cues and regions in the video using the ''::cue'', ''::cue()'', ''::cue-region'' and
 ''::cue-region()'' pseudo-elements.</p>
 
+<p>Only a limited set of CSS styles is allowed in WebVTT because user agents that do not support CSS
+will need to implement CSS functionality equivalents. Browsers therefore also have to be limited in
+what CSS styles they support for WebVTT so as to enable identical rendering in browsers and
+non-browsers.</p>
```
Comment (nigelmegitt) [inline index.bs]: I am not clear that this is a genuine requirement, that rendering is identical in browsers and non-browsers. If I recall correctly WebVTT has no requirements document. Pointer to a WG resolution that this is important?

Comment (dwsinger) [reply index.bs]: Actually this sentence is subtly wrong; user agents that don't support CSS (a conformance class as Andreas notes), well, do not support CSS.  UAs that DO support CSS *can* use a full CSS engine but can also implement this restricted subset of CSS.

Comment (silviapfeiffer) [reply index.bs]: OK, I'll reformulate.

**Comment by silviapfeiffer** [comment]
review again :-)

File: index.bs [hunk 2]
```diff
@@ -307,9 +307,9 @@ target WebVTT cues and regions in the video using the ''::cue'', ''::cue()'', ''
 ''::cue-region()'' pseudo-elements.</p>
 
 <p>Only a limited set of CSS styles is allowed in WebVTT because user agents that do not support CSS
-will need to implement CSS functionality equivalents. Browsers therefore also have to be limited in
-what CSS styles they support for WebVTT so as to enable identical rendering in browsers and
-non-browsers.</p>
+will need to implement CSS functionality equivalents. User agents that support a full CSS engine
+must limited the CSS styles they interpret for WebVTT so as to enable identical rendering without
```
Comment (nigelmegitt) [inline index.bs]: It's not clear what the "identical rendering" is referring to here - identical to what?

**Comment by dwsinger** [comment]
I think this is still wrong. User agents that don't support CSS don't support CSS, that's why we had to do the colour work.  This is about UAs that DO support CSS but DON'T have a full engine; they are only required to support this subset of CSS. Please stay silent on what happens if you have a full engine; I think requiring those implementations to be sure to limit themselves to that subset is adding unneeded pain.

File: index.bs (hunk 2 above)
Comment (silviapfeiffer) [reply index.bs]: Identical to what is rendered when you follow the spec and have CSS functionality.

**Comment by silviapfeiffer** [comment]
@dwsinger No, this paragraph is not about UAs that support CSS partially. I don't think such a class exists. It's about the difference between UAs that support CSS and those that don't support CSS (like VLC). It says that those that support CSS (like Web browsers) must not allow the full capability of their CSS support in WebVTT and **this is something I am not going to budge on**. It is extremely important that we disallow full engines to allow all their functionality in WebVTT, because if we do, then everyone has to implement the full CSS engine for WebVTT.

Remember the times when Internet Explorer was the dominant Web browser and they implemented functionality that was not in the HTML&CSS standards? What did that do to the other browsers? This paragraph is about avoiding that situation, so we get compatible implementations.

**Comment by dwsinger** [comment]
OK, there are two problems here, and we should disentangle them. (a) your text talks about non-CSS UAs that only implement a subset. Those are NOT non-CSS UAs, non-CSS UAs don't implement any CSS. (b) then there is the question of what CSS you are allowed to use and the UA is required to support, and whether the UA has to ignore anything else. I'm OK with a parse-time discard on (b). So, the text needs to talk about CSS-supporting UAs only supporting the CSS we document.

**Comment by silviapfeiffer** [comment]
@dwsinger re (a) Read that sentence properly please. It says:

"Only a limited set of CSS styles is allowed in WebVTT because user agents that do not support CSS
will need to implement CSS functionality equivalents."

this is referring to:
1/ the WebVTT spec specifying a limited set of CSS styles
2/ UAs that don't support CSS
3/ CSS functionality **equivalents**

It does not say that these UAs implement a subpart of CSS. It says they have to implement functionality **equivalents** of the limited CSS subset defined in WebVTT.

ad (c) The second sentence says exactly what you are suggesting.

Could you please provide proposed wording changes that make it clearer for you? I really don't know how else to say it more clearly.

**Comment by silviapfeiffer** [comment]
OK, sounds to me like we need to define our conformance classes better: one that has no CSS support, one that supports the CSS parts that WebVTT uses (see #391) and UAs with a full CSS engine.

**Comment by silviapfeiffer** [comment]
Is also relate to #389

**Comment by dwsinger** [comment]
possibly, but I think you argued that the reduced CSS and full CSS engines had to give the same result, so probably only two

**Comment by silviapfeiffer** [comment]
Not quite. "WebVTT CSS" supporting UAs behave differently, since they don't follow the rendering section to the word. So, it needs an out from some of the spec. I think we need three levels.

1/ No CSS: non-CSS UAs that ignore all CSS

2/ WebVTT CSS: non-CSS UAs that implement functional equivalents to the WebVTT CSS features and therefore are allowed to not follow the text of the specification in the rendering section

3/ Full CSS: CSS-supporting UAs (mostly browsers) that implement the WebVTT CSS functionality as specified

**Comment by silviapfeiffer** [comment]
Please review #389 . If that one's ok, then we can move forward to reformulate the paragraph here and base it on those three conformance classes. (Incidentally, the third one is not explicitly formulated in #389 because it's implied by the spec.)

**Comment by silviapfeiffer** [comment]
Since #389 landed, I have now updated this PR. I decided to move the paragraph out of an example section into a normative space and it made sense to explicitly define the "Full CSS engine" conformance class for this. Please re-review.

File: index.bs [hunk 3]
```diff
@@ -765,6 +760,13 @@ requirements.</p>
  [[#parsing]], [[#rendering]] and [[#css-extensions]] in such a way that the rendered results are
  equivalent to what a full CSS supporting renderer produces.</p></dd>
 
+ <dt><dfn>User agents that support a full HTML CSS engine</dfn></dt>
+ <dd><p>All processing requirements in this specification apply. However, only a limited set of CSS
+ styles is allowed because <a>user agents that do not support a full HTML CSS engine</a> will need
+ to implement CSS functionality equivalents. User agents that support a full CSS engine must
+ therefore limited the CSS styles they interpret for WebVTT so as to enable identical rendering
```
Comment (nigelmegitt) [inline index.bs]: typo: s/limited/limit

Is "interpret" the right term here, addressing your concerns @dwsinger ? Seems to me that "apply" would be better.

Comment (dwsinger) [reply index.bs]: yes, s/limited/limit/ and "limit the CSS they apply" might be better

Comment (silviapfeiffer) [reply index.bs]: ok

**Comment by silviapfeiffer** [comment]
Please re-review

**Comment by dwsinger** [comment]
I'm (still) OK

**Pull request merged**